### DISCUSSION TITLE
[Feature] Smart Immich Assistant: A Wild Idea?

### DISCUSSION THREAD

**Discussion by radh21301** [immich-app/immich#11110] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

This feels like a far fetched thing, but seems like both the users and the contributors would benefit.

**The problem:**
I have been in situations where I asked for something or mentioned an issue, and was then directed to an already existing post on github. In many cases, it is because github search does not really understand context, and it seems to use something like key words -> so even though the OP did their due diligence, it doesn't really help. It is very easier for anyone to raise a new post on github, but I imagine it is time consuming for the devs to find similar posts and then close the current post referencing the existing one. It is also inefficient for the one creating the post because it might take a while for the devs to respond.

**A possible solution:**
A chat assistant (something like [Immich Assistant](https://chat.openai.com/g/g-aqL2eoc8I-immich-assistant) ChatGPT extension) but something that sits right here in github. The model has all the information present in the discussions, issues, immich docs etc.
The way I imagine it working:
* Someone makes a post and hits submit
* The bot then uses the model to check if there are similar posts
* It responds to the post asking "Hey OP, do these post seem similar to you?"
* If the OP thinks, it is similar, they can close it. 
* Else, the post stays and the devs can choose to respond.

**Other use cases**
Since this model has access to all latest information immich has, it could definitely be made into a chat assistant (on the immich website) where users can ask it questions, get answers (or directed to the page/post which has the info) etc. Similar to the the ChatGPT assistant, but sits right here in github/immich website. I think this might help people easily find answers, and in turn reduce the number of questions on discord. What if discord data can also be fed to the model?-> the possibilities become endless.

Is something like this even possible? If it is, is it feasible to implement? I am just spitballing here. I have no software background - just though this would be a cool thing to have.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by mertalev** [comment]
We've wanted to do some kind of smart deduplication for some time. Having a bot that suggests similar issues or discussions seems like a really nice idea!

**Comment by aviv926** [comment]
That's could be really nice to have
